### PR TITLE
Fix the ``aggregation`` argument of ``SamplingVQE``

### PR DESCRIPTION
**Pull request by Cryoris**
### Summary

In the `SamplingVQE` the aggregation was not passed to the `_DiagonalEstimator` and therefore had no effect! 
Also fixed  a bug in the CVaR aggregation function, which overshot the expected value and improved the docs to include an explanation how a custom aggregation should look like.

Thanks @t-imamichi for finding the bug!

### CODE REVIEW

File: qiskit/algorithms/minimum_eigensolvers/diagonal_estimator.py [hunk 1]
```diff
@@ -171,8 +171,9 @@ def aggregate(measurements):
             accumulated_percent = 0  # once alpha is reached, stop
             cvar = 0
             for probability, value in sorted_measurements:
-                cvar += value * max(probability, alpha - accumulated_percent)
+                cvar += value * min(probability, alpha - accumulated_percent)
                 accumulated_percent += probability
+                print(value, probability, accumulated_percent, cvar)
```
Comment (t-imamichi) [inline qiskit/algorithms/minimum_eigensolvers/diagonal_estimator.py:176]:
```suggestion
```

**Comment by t-imamichi** [comment]
Thank you for the fix!

File: qiskit/algorithms/minimum_eigensolvers/diagonal_estimator.py (hunk 1 above)
Comment (woodsp-ibm) [reply qiskit/algorithms/minimum_eigensolvers/diagonal_estimator.py:176]: If something like this is useful we could always have a logging statement here instead that does similar

Comment (Cryoris) [reply qiskit/algorithms/minimum_eigensolvers/diagonal_estimator.py:176]: whoops -- I thought I removed all statements!

Comment (Cryoris) [reply qiskit/algorithms/minimum_eigensolvers/diagonal_estimator.py]: @woodsp-ibm That was for debugging purposes only 🙂

**Review by woodsp-ibm — APPROVED** [review]

**Pull request merged**